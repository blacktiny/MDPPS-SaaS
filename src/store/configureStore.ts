import { createStore, applyMiddleware } from 'redux'
import { persistStore, persistReducer } from 'redux-persist'
import { composeWithDevTools } from 'redux-devtools-extension'
import storage from 'redux-persist/lib/storage'

import rootReducer from './reducer'
import { PersistConfig } from 'redux-persist'
import { RootState } from './types'

const persistConfig: PersistConfig<RootState> = {
  key: 'root',
  storage: storage,
  blacklist: ['options']
}

const persistedReducer = persistReducer(persistConfig, rootReducer)
const store = createStore(persistedReducer, composeWithDevTools(applyMiddleware()))

export const persistor = persistStore(store)

const configureStore = () => ({
  ...store
})

export default configureStore()
