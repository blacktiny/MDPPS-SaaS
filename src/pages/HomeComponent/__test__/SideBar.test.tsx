import { cleanup, render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import { Store } from 'redux'
import configureStore from 'redux-mock-store'
import MockAxios from '../../../__mocks__/axios'
import SideBarComponent from '../SideBar'
// import { navigate } from '@reach/router'
let mockStore: Store

jest.mock('@reach/router', () => ({
  navigate: jest.fn()
}))

beforeEach(() => {
  MockAxios.reset()
  MockAxios.mockClear()
  mockStore = configureStore()({
    auth: {
      user: {
        account_type: 'brand',
        active_company: {
          id: 5324,
          url: 'https://dev.mdpps.com/api/companies/5324/',
          business_name: 'Softsuave 2',
          microsite_name: 'softsuave-2-2',
          permissions: [
            'brand_manager.view_brand',
            'adn.view_dealerapplicationconfiguration',
            'analytics.view_integration',
            'Authenticated',
            'accounts.view_powerprofile'
          ],
          is_subscribed: false,
          status: 'pending_review'
        }
      }
    }
  })
})
afterEach(() => {
  cleanup()
})
test('Side bar is loaded', async () => {
  const { getByTestId } = render(
    <Provider store={mockStore}>
      <SideBarComponent onSideBarClose={() => {}} />
    </Provider>
  )
  expect(getByTestId).toMatchSnapshot()
})
// test('Integration popup is opened', async () => {
//   const { getByTestId } = render(
//     <Provider store={mockStore}>
//       <SideBarComponent onSideBarClose={() => {}} />
//     </Provider>
//   )
//   act(() => {
//     getByTestId('button_integrations').click()
//   })
//   expect(getByTestId('button_integrations')).toHaveAttribute(
//     'data-testid',
//     'button_integrations'
//   )
// })
// test('Marketplace submenu will be open in the sidebar', async () => {
//   const { getByTestId } = render(
//     <Provider store={mockStore}>
//       <SideBarComponent onSideBarClose={() => {}} />
//     </Provider>
//   )
//   act(() => {
//     getByTestId('button_marketplace').click()
//   })
//   expect(getByTestId('Brand')).toHaveAttribute('data-testid', 'Brand')
// })

// test('Should redirect to correct path if user click on Sign up button', async () => {
//   const { getByTestId } = render(
//     <Provider store={mockStore}>
//       <SideBarComponent onSideBarClose={() => {}} />
//     </Provider>
//   )
//   act(() => {
//     getByTestId('button_myBrands').click()
//   })

//   expect(navigate).toHaveBeenCalledTimes(1)
//   expect(navigate).toHaveBeenCalledWith('/dashboard/home/in-progress')
// })
