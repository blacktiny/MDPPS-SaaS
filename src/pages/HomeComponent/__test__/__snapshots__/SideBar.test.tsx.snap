// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side bar is loaded 1`] = `[Function]`;
